src/containerapp/azext_containerapp/containerapp_resiliency_decorator.py

```diff
@@ -0,0 +1,484 @@
+
+        self.containerapp_resiliency_def["properties"]["httpRetryPolicy"] = http_retry_def
+        self.containerapp_resiliency_def["properties"]["timeoutPolicy"] = timeout_def
+        self.containerapp_resiliency_def["properties"]["tcpRetryPolicy"] = tcpretry_def
+        self.containerapp_resiliency_def["properties"]["circuitBreakerPolicy"] = circuitbreaker_def
+        self.containerapp_resiliency_def["properties"]["tcpConnectionPool"] = tcp_connectionpool_def
+        self.containerapp_resiliency_def["properties"]["httpConnectionPool"] = http_connectionpool_def
+
+        self.containerapp_resiliency_def = clean_null_values(self.containerapp_resiliency_def)
+
+        if self.containerapp_resiliency_def is None or self.containerapp_resiliency_def == {}:
+            self.containerapp_resiliency_def["properties"] = {}
+    
+    def set_up_default_containerapp_resiliency(self):
+        if self.get_argument_tcp_retry_max_attempts() or self.get_argument_cb_consecutive_errors()\
+                or self.get_argument_cb_interval() or self.get_argument_cb_max_ejection() or \
+                self.get_argument_tcp_max_connections() or self.get_argument_http1_max_pending_req() \
+                or self.get_argument_http2_max_req() or self.get_argument_timeout_response() \
+                or self.get_argument_timeout_connection() or self.get_argument_http_retry_max() \
+                or self.get_argument_http_retry_delay() or self.get_argument_http_retry_interval() \
+                or self.get_argument_http_retry_codes() or self.get_argument_http_retry_errors():
+            not self.get_argument_disable_warnings() and logger.warning(
+                'Additional flags were passed along with --default. These flags will be ignored, and system defaults will be applied.')
+        
+        timeout_def = TimeoutPolicyModel
+        timeout_def["responseTimeoutInSeconds"] = DEFAULT_RESPONSE_TIMEOUT
+        timeout_def["connectionTimeoutInSeconds"] = DEFAULT_CONNECTION_TIMEOUT
+        self.containerapp_resiliency_def["properties"]["timeoutPolicy"] = timeout_def
+
+        tcpretry_def = TcpRetryPolicyModel
+        tcpretry_def["maxConnectAttempts"] = DEFAULT_TCP_MAX_CONNECT_ATTEMPTS
+        self.containerapp_resiliency_def["properties"]["tcpRetryPolicy"] = tcpretry_def
+
+        circuitbreaker_def = CircuitBreakerPolicyModel
+        circuitbreaker_def["consecutiveErrors"] = DEFAULT_CONSECUTIVE_ERRORS
+        circuitbreaker_def["intervalInSeconds"] = DEFAULT_INTERVAL
+        circuitbreaker_def["maxEjectionPercent"] = 100
+        self.containerapp_resiliency_def["properties"]["circuitBreakerPolicy"] = circuitbreaker_def
+
+        http_retry_policy = {
+            "maxRetries": 3,
+            "retryBackOff": {
+                "initialDelayInMilliseconds": 1000,
+                "maxIntervalInMilliseconds": 10000
```

Comment:
I think there are constants for these, right?

Reply:
Fixed